src/TagHeaderFormatter/MaxHeaderValueLengthFormatter.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/*
+ * This file is part of the FOSHttpCache package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\HttpCache\TagHeaderFormatter;
+
+use FOS\HttpCache\Exception\InvalidTagException;
+
+/**
+ * A header formatter that splits the value(s) from a given
+ * other header formatter (e.g. the CommaSeparatedTagHeaderFormatter)
+ * into multiple headers making sure none of the header values
+ * exceeds the configured limit.
+ *
+ * @author Yanick Witschi <yanick.witschi@terminal42.ch>
+ */
+class MaxHeaderValueLengthFormatter implements TagHeaderFormatter
+{
+    /**
+     * @var TagHeaderFormatter
+     */
+    private $inner;
+
+    /**
+     * @var int
+     */
+    private $maxHeaderValueLength;
+
+    /**
+     * The default value of the maximum header length is 4096 because most
+     * servers limit header values to 4kb.
+     * HTTP messages cannot carry characters outside the ISO-8859-1 standard so they all
+     * use up just one byte.
+     *
+     * @param TagHeaderFormatter $inner
+     * @param int                $maxHeaderValueLength
+     */
+    public function __construct(TagHeaderFormatter $inner, $maxHeaderValueLength = 4096)
+    {
+        $this->inner = $inner;
+        $this->maxHeaderValueLength = $maxHeaderValueLength;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTagsHeaderName()
+    {
+        $this->inner->getTagsHeaderName();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTagsHeaderValue(array $tags)
+    {
+        $values = (array) $this->inner->getTagsHeaderValue($tags);
+        $newValues = [[]];
```

Comment:
should this not be a single `[]`?

Reply:
Nope, this is because of `array_merge(...$newValues);`. You could also call `array_merge()` in every loop but it's a performance gain if you only merge it once :)